google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseInfo.java

```diff
@@ -141,6 +155,29 @@ public Builder setDialect(Dialect dialect) {
       return this;
     }
 
+    @Override
+    public Builder setProtoDescriptors(@Nonnull byte[] protoDescriptors) {
+      Preconditions.checkNotNull(protoDescriptors);
+      this.protoDescriptors = ByteString.copyFrom(protoDescriptors);
+      return this;
+    }
+
+    @Override
+    public Builder setProtoDescriptors(@Nonnull InputStream inputStream) throws IOException {
+      Preconditions.checkNotNull(inputStream);
+      byte[] byteArray = ByteStreams.toByteArray(inputStream);
+      this.protoDescriptors = ByteString.copyFrom(byteArray);
+      return this;
+    }
+
+    @Override
+    public Builder setProtoDescriptors(String filePath) throws IOException {
+      Preconditions.checkState(filePath.length() != 0, "Input File Path cannot be empty.");
+      InputStream inputStream = getClass().getClassLoader().getResourceAsStream(filePath);
+      Preconditions.checkNotNull(inputStream, "Input File path is invalid.");
+      return setProtoDescriptors(inputStream);
```

Comment:
```suggestion
      return setProtoDescriptors(inputStream);
```
`public Builder setProtoDescriptors(@Nonnull InputStream inputStream)` already does the NotNull check, so we need not have a redundant check here.

Reply:
@rajatbhatta 
When the input file path is not valid, then `getResourceAsStream` does not throw any exception instead `InputStream` becomes null. If we don't make a check here then it calls `public Builder setProtoDescriptors(@Nonnull InputStream inputStream)` which has a `NotNull` check, but the user gets an error that `InputStream cannot be null` which is not expected as the user is passing a file path as String.
Instead the user should know that the Input File Path is invalid and hence we might need this check here.